File: Segundo/Monopoly/civitas_ruby (1)/proyecto_civitas_ruby/lib/sorpresa_por_casa_hotel.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module Civitas
class SorpresaPorCasaHotel < Sorpresa
  def initialize(texto,valor)
    super(texto)
    @valor = valor
  end
  
  def aplicarAJugador(actual,todos)
    if(jugadorCorrecto(actual,todos))
        super(actual,todos)
        todos[actual].modificarSaldo(@valor*todos[actual].cantidadCasasHoteles)
    end
  end
  
  def toString
    a = "Por Casa-Hotel"
    return a
  end
  public_class_method :new
end
end

File: Segundo/Monopoly/civitas_java (1)/proyecto_civitas/src/civitas/SorpresaPorCasaHotel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;
import java.util.ArrayList;
/**
 *
 * @author pc
 */
public class SorpresaPorCasaHotel extends Sorpresa{
    private int valor;

    SorpresaPorCasaHotel(int valor,String texto)
    {
        super(texto);
        this.valor=valor;
    }
    
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual,todos))
        {
            informe(actual,todos);
            todos.get(actual).modificarSaldo(valor*todos.get(actual).cantidadCasasHoteles());
        }
    }

    @Override
    public String toString() 
    {
        return "Sorpresa Por CasaHotel: " + "valor: " + valor +super.toString() +'\n';
    }
}
